packages/modules/clickhouse/src/clickhouse-container.test.ts

```diff
@@ -0,0 +1,184 @@
+        url: container.getHttpUrl(),
+        username: container.getUsername(),
+        password: container.getPassword(),
+      });
+
+      const result = await client.query({
+        query: "SELECT 1 AS value",
+        format: "JSON",
+      });
+
+      const data = (await result.json()) as ClickHouseQueryResponse<{ value: number }>;
+      expect(data?.data?.[0]?.value).toBe(1);
+    } finally {
+      await client?.close();
+      await container.stop();
+    }
+  });
+  // }
+
+  // setDatabase {
+  it("should set database", async () => {
+    const customDatabase = "customDatabase";
+    const container = await new ClickHouseContainer().withDatabase(customDatabase).start();
+
+    let client: ClickHouseClient | undefined;
+
+    try {
+      client = createClient(container.getClientOptions());
+
+      const result = await client.query({
+        query: "SELECT currentDatabase() AS current_database",
+        format: "JSON",
+      });
+
+      const data = (await result.json()) as ClickHouseQueryResponse<{ current_database: string }>;
+      expect(data?.data?.[0]?.current_database).toBe(customDatabase);
+    } finally {
+      await client?.close();
+      await container.stop();
+    }
+  });
+  // }
+
+  // setUsername {
+  it("should set username", async () => {
+    const customUsername = "customUsername";
+    const container = await new ClickHouseContainer().withUsername(customUsername).start();
+
+    let client: ClickHouseClient | undefined;
+
+    try {
+      client = createClient(container.getClientOptions());
+
+      const result = await client.query({
+        query: "SELECT currentUser() AS current_user",
+        format: "JSON",
+      });
+
+      const data = (await result.json()) as ClickHouseQueryResponse<{ current_user: string }>;
+      expect(data?.data?.[0]?.current_user).toBe(customUsername);
+    } finally {
+      await client?.close();
+      await container.stop();
+    }
+  });
+  // }
+
+  it("should work with restarted container", async () => {
+    const container = await new ClickHouseContainer().start();
+    await container.restart();
+
+    let client: ClickHouseClient | undefined;
+    try {
+      client = createClient(container.getClientOptions());
+
+      const result = await client.query({
+        query: "SELECT 1 AS value",
+        format: "JSON",
+      });
+
+      const data = (await result.json()) as ClickHouseQueryResponse<{ value: number }>;
+      expect(data?.data?.[0]?.value).toBe(1);
+    } finally {
+      await client?.close();
+      await container.stop();
+    }
+  });
+
+  /**
+   * Verifies that a custom Docker health check that fails immediately
+   * causes the container startup process (`container.start()`) to reject with an error.
+   *
+   * Note: This test pattern was adapted from a similar test case used for
+   * PostgreSQL containers to ensure custom failing health checks are handled correctly.
+   */
+  it("should allow custom healthcheck", async () => {
```

Comment:
This was only necessary when we used the health check wait strategy, to ensure we didn't overwrite a user's. Now that we use the `HttpWaitStrategy`, there's no risk of us conflicting health check definitions. Can be removed

Reply:
Thanks for explaining, I must admit I didn't 100% understand the purpose of that test 😅 Removed it.